### ISSUE TITLE
_.merge modifies source objects

### ISSUE THREAD

**Issue by jcope2013** [lodash/lodash#1987]
similar to https://github.com/lodash/lodash/issues/1801

```
var a = {predicates: [{a: 1}]};
var b = {predicates: [{b: 2}]};
var c = _.merge({}, a);
var d = [b].reduce(function(config, addon) {
    return _.merge(config, addon);
}, c);
console.log("a.predicates: "+JSON.stringify(a.predicates));
console.log("c.predicates: "+JSON.stringify(b.predicates));
console.log("c.predicates: "+JSON.stringify(c.predicates));
console.log("d.predicates: "+JSON.stringify(d.predicates));
```

3.10.1 (https://jsfiddle.net/w3nk2d24/)

```
a.predicates: [{"a":1}]
b.predicates: [{"b":2}]
c.predicates: [{"a":1,"b":2}]
d.predicates: [{"a":1,"b":2}]
```

4.3.0 (https://jsfiddle.net/pctkkc4c/)
```
a.predicates: [{"a":1,"b":2}]
b.predicates: [{"b":2}]
c.predicates: [{"a":1,"b":2}]
d.predicates: [{"a":1,"b":2}]
```

I would instead expect the value of a.predicates to be unchanged. What do you think?

**Comment by jdalton** [comment]
Can you convert these to jsbins. It's hard to parse out the actual source of a jsfiddle (it's hidden away)

**Comment by jcope2013** [comment]
4.3 (https://jsbin.com/naqewafefi/1/edit?html,js,output)

3.10.1 ( https://jsbin.com/miricimeyi/1/edit?html,js,output)

**Comment by jdalton** [comment]
Thank you!

**Comment by janmeier** [comment]
@jdalton This commit is causing a bit of trouble for us in sequelize - care to weigh in on https://github.com/sequelize/sequelize/pull/5438?
